packages/grafana-ui/src/components/uPlot/utils.ts

```diff
@@ -181,6 +181,11 @@ export function findMidPointYPosition(u: uPlot, idx: number) {
     y = u.valToPos((min || max)!, u.series[(sMaxIdx || sMinIdx)!].scale!);
   }
 
+  // if y is out of canvas bounds, snap it to the bottom
+  if (y !== undefined && y < 0) {
+    y = u.bbox.height;
```

Comment:
`bbox` is in canvas pixels, so i think this needs a `/ devicePixelRatio`

(`u.valToPos()` returns CSS pixels, unless third arg is `true` for canvas pixels)

Reply:
You are right this needed the devicePixelRatio to work properly.